### DISCUSSION TITLE
ZeroSync Bitcoin Light Client on Ethereum

### DISCUSSION THREAD

**Discussion by orenyomtov** [ZeroSync/ZeroSync#68] [Ideas]
There have been many projects like
http://btcrelay.org/
https://bitcoinmirror.org/

That attempted to bring a bitcoin light client to ethereum.

They all stopped updating the block headers, thus rendering them broken, because of the high gas costs it takes to run a TX to update the block headers every ±10 minutes (the gas costs around half a million dollars a year in my back of the envelope estimation).

I believe a succinct-ZK-based solution like ZeroSync can solve this issue by not requiring this constant updating, and instead, every actor that wants to prove state of the bitcoin chain (like transaction inclusion) can bring their own ZeroSync proof and validate it (hopefully for less than 500k gas, meaning ±10$).

I'd love to hear your thoughts on the idea, or even better, if you already have "creating a ZeroSync solidity verifier" scheduled on the roadmap.

**Comment by lucidLuckylee** [comment]
The project originated from exactly this idea of a zero knowledge bitcoin relay to save gas costs. Submtting a STARK proof to ethereum is quite costly though (about 300$ according to https://twitter.com/andrewmilson/status/1686292241990692864).
We have a recursive proof of the first 800k block headers ready and that could be submitted to ethereum. To keep up with the chain afterwards one is better off going through an Ethereum rollup to batch proofs together and reduce the cost.
As far as I am aware all of that should be possible right now with e.g. Starknet it would just be a bit of a hassle to get the initial proof submitted to Ethereum as with sandstorm there seemed to be a lot of technicalities involved and StarkWare's shared prover (SHARP) did not allow a computation larger than 1-2 Million steps. If we were to just recursively verify the header chain proof for the first 800k headers it is a bit more than that at the moment.
